kernel/src/engine/default/parquet.rs

```diff
@@ -245,6 +247,63 @@ impl<E: TaskExecutor> ParquetHandler for DefaultParquetHandler<E> {
             self.readahead,
         )
     }
+
+    /// Writes engine data to a Parquet file at the specified location.
+    ///
+    /// This implementation uses asynchronous file I/O with object_store to write the Parquet file.
+    /// If a file already exists at the given location, it will be overwritten.
+    ///
+    /// # Parameters
+    ///
+    /// - `location` - The full URL path where the Parquet file should be written
+    ///   (e.g., `s3://bucket/path/file.parquet`, `file:///path/to/file.parquet`).
+    /// - `data` - An iterator of engine data to write to the Parquet file.
+    ///
+    /// # Returns
+    ///
+    /// A [`DeltaResult`] containing [`FileMeta`] with the file's location, size, and modification time.
+    fn write_parquet_file(
+        &self,
+        location: url::Url,
+        mut data: Box<dyn Iterator<Item = DeltaResult<Box<dyn EngineData>>> + Send>,
+    ) -> DeltaResult<FileMeta> {
+        let path = Path::from_url_path(location.path())?;
+        let store = self.store.clone();
+
+        // Get first batch to initialize writer with schema
+        let first_batch = data.next().ok_or_else(|| {
+            Error::generic("Cannot write parquet file with empty data iterator")
+        })??;
+        let first_arrow = ArrowEngineData::try_from_engine_data(first_batch)?;
+        let first_record_batch: RecordBatch = (*first_arrow).into();
+
+        // Collect remaining batches
+        let mut batches = vec![first_record_batch];
+        for result in data {
+            let engine_data = result?;
+            let arrow_data = ArrowEngineData::try_from_engine_data(engine_data)?;
+            let batch: RecordBatch = (*arrow_data).into();
+            batches.push(batch);
+        }
```

Comment:
Could we not move all this into the async block? Then we don't have to collect into a vec, we can just get the schema out from the first data and the loop over the rest of the batches (turning them into arrow as you go)

Ditto below